pkg/kubelet/kuberuntime/kuberuntime_image.go

```diff
@@ -96,6 +96,18 @@ func (m *kubeGenericRuntimeManager) GetImageRef(ctx context.Context, image kubec
 	return resp.Image.Id, nil
 }
 
+func (m *kubeGenericRuntimeManager) GetImageSize(ctx context.Context, image kubecontainer.ImageSpec) (uint64, error) {
```

Comment:
one question we were discussing in the past is whether this size is compressed or uncompressed. And also whether it will be consistent across runtimes.

Reply:
cc @haircommander was it you we discussed compressed vs. uncompressed with?

Reply:
For containerd it's compressed: https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/pkg/cri/store/image/image.go#L47

I think compressed makes sense for this purpose because it's the actual size that's transfered over the network.

Reply:
another note is it doesn't give a good representation of shared layers AFAICS. Even if it's uncompressed two images' sizes will double count a shared layer

Reply:
yeah, this is less of a problem here (for the purpose of metric reporting), but more of a problem on image GC. kubelet's image GC relies on image.Size to determine how much space is freed, which might not be accurate considering decompression, shared layers, etc.

Reply:
I'm actually having trouble figuring out whether it's compressed or uncompressed in cri-o but one source shows uncompressed

Reply:
+1 on compressed for the image pulling case.